lib/src/wallet/transactions.rs

```diff
@@ -199,6 +184,35 @@ impl TransactionMetadataSet {
                 }
             })
         });
+        self.remove_domain_specific_txids::<SaplingDomain<Network>>(&txids_to_remove);
+        self.remove_domain_specific_txids::<OrchardDomain>(&txids_to_remove);
+    }
+
+    fn remove_domain_specific_txids<D: DomainWalletExt<Network>>(
```

Comment:
This is internal logic in the call stack that transforms a spent note into an unspent note.   We'll revisit naming.

Reply:
Tracked here:  https://github.com/zingolabs/zingolib/issues/114#issuecomment-1260191887